test/app/lib/encoding/slip32.spec.ts

```diff
@@ -0,0 +1,129 @@
+        const {key, chaincode} = getBytesFromHex(item.prv_hex)
+        expect(importedKey.extendedKey.key.bytes).toEqual(key)
+        expect(importedKey.extendedKey.chainCode).toEqual(chaincode)
+      })
+    })
+  })
+describe(`Slip32 key exchange: import public key (${fixtures.slip32.valid.length} test vectors)`,
+  () => {
+    fixtures.slip32.valid.forEach((item, index) => {
+      it("should import public key from slip32 " +
+        `(${index + 1}, key: ${cropString(item.pub_bech32, 20, true)})`, () => {
+        const importedKey = exchange.importKeyFromSlip32(item.pub_bech32)
+        // Compare key path
+        const keyPath = KeyPath.fromString(item.keyPath)
+        expect(importedKey.keyPath).toEqual(keyPath)
+        // Compare hex to imported key data
+        const {key, chaincode} = getBytesFromHex(item.pub_hex)
+        expect(importedKey.extendedKey.key.bytes).toEqual(key)
+        expect(importedKey.extendedKey.chainCode).toEqual(chaincode)
+      })
+    })
+  })
+describe(`Slip32 key exchange: export private key (${fixtures.slip32.valid.length} test vectors)`,
+  () => {
+    fixtures.slip32.valid.forEach((item, index) => {
+      it("should export private key as slip32 " +
+        `(${index + 1}, hex: ${cropString(item.prv_hex, 20)})`, () => {
+        const keyPath = KeyPath.fromString(item.keyPath)
+        const {key, chaincode} = getBytesFromHex(item.prv_hex)
+        const extendedKey: Key.ExtendedKey<PrivateKey.PrivateKey> =
+          PrivateKey.extend(PrivateKey.fromBytes(key), chaincode)
+        const exportedKey = exchange.exportKeyToSlip32(keyPath, extendedKey)
+        expect(exportedKey).toEqual(item.prv_bech32)
+      })
+    })
+  })
+describe(`Slip32 key exchange: export public key (${fixtures.slip32.valid.length} test vectors)`,
+  () => {
+    fixtures.slip32.valid.forEach((item, index) => {
+      it("should export public key as slip32 " +
+        `(${index + 1}, hex: ${cropString(item.pub_hex, 20)})`, () => {
+        const keyPath = KeyPath.fromString(item.keyPath)
+        const {key, chaincode} = getBytesFromHex(item.pub_hex)
+        const extendedKey: Key.ExtendedKey<PublicKey.PublicKey> =
+          PublicKey.extend(PublicKey.fromBytes(key), chaincode)
+        const exportedKey = exchange.exportKeyToSlip32(keyPath, extendedKey)
+        expect(exportedKey).toEqual(item.pub_bech32)
+      })
+    })
+  })
+
+/**
+ * Get key and chaincode bytes from hexidecimal representation
+ * @param {string} hex
+ * @returns {{key: Buffer; chaincode: Buffer}}
+ */
+function getBytesFromHex(hex: string) {
+  const buffer = Buffer.from(hex, "hex")
+  const key = buffer.slice(buffer.length - 33)
+  const chaincode = buffer.slice(buffer.length - 33 - 32, buffer.length - 33)
+
+  return {key, chaincode}
+}
+
+/**
+ * Crop long strings to be displayed in the tests
+ * @param {string} text
+ * @returns {string}
+ */
+function cropString(text: string, limit: number, end?: boolean) {
+  if (!end) {
```

Comment:
`end` could use an explanation on what is it for. AFAICT, it tells the function to crop by the beginning (default) or end of the string, but that's not so obvious from the code itself :sweat_smile: